cpp/ql/src/semmle/code/cpp/ir/implementation/raw/internal/TranslatedCall.qll

```diff
@@ -374,6 +374,35 @@ class TranslatedSideEffects extends TranslatedElement, TTranslatedSideEffects {
   override Function getFunction() { result = expr.getEnclosingFunction() }
 }
 
+class TranslatedStructorCallSideEffects extends TranslatedSideEffects {
+  TranslatedStructorCallSideEffects() { getParent().(TranslatedStructorCall).hasQualifier() }
+
+  override predicate hasInstruction(Opcode opcode, InstructionTag tag, Type t, boolean isGLValue) {
+    opcode instanceof Opcode::IndirectMustWriteSideEffect and
```

Comment:
Why "must"? Is that the same as saying that all constructors must write to a field? Or to _every_ field?

Reply:
That would be to every field - it should probably be "may", but I think we'd need to make sure to have an uninitialized value for every constructed object.

Reply:
I believe we do have a definition to pre-initialize every constructed object. For a stack object, there will be an explicit `Uninitialized`. For a heap object, the call to the allocator will have a side effect that represents the initial (possibly uninitialized) value of the memory.